tiledb/api/src/datatype/mod.rs

```diff
@@ -1,194 +1,34 @@
-pub mod logical;
-pub mod physical;
-
-pub use logical::*;
-pub use physical::{PhysicalType, PhysicalValue};
+#[cfg(feature = "arrow")]
+pub mod arrow;
 
-use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+use crate::{FromStringCore, ToStringCore};
 
-use serde::{Deserialize, Serialize};
-use util::option::OptionSubset;
+pub use tiledb_common::datatype::*;
 
-use crate::error::DatatypeErrorKind;
-use crate::Result as TileDBResult;
+impl ToStringCore for Datatype {
```

Comment:
The enum is moved to `tiledb-common/src/datatype/mod.rs`.

What's here now is just trait impls for new traits `ToStringCore` and `FromStringCore` which call the tiledb core string functions.

Since these require linking with libtiledb, `tiledb-common` specifically does not want them.  Hence Debug is derived and Display just uses that.